packages/two_dimensional_scrollables/lib/src/tree_view/tree.dart

```diff
@@ -0,0 +1,1046 @@
+  /// by [TreeViewNode.depth]. This allows for more customization in building
+  /// tree rows, such as filling the indented area with decorations or ink
+  /// effects.
+  final TreeViewIndentationType indentation;
+
+  /// A wrapper method for triggering the expansion or collapse of a
+  /// [TreeViewNode].
+  ///
+  /// Use as part of [TreeView.defaultTreeNodeBuilder] to wrap the leading icon
+  /// of parent [TreeViewNode]s such that tapping on it triggers the animation.
+  ///
+  /// If defining your own [TreeView.treeNodeBuilder], this method can be used
+  /// to wrap any part, or all, of the returned widget in order to trigger the
+  /// change in state for the node when tapped.
+  ///
+  /// The gesture uses [HitTestBehavior.translucent], so as to not conflict
+  /// with any [TreeRow.recognizerFactories] or other interactive content in the
+  /// [TreeRow].
+  static Widget toggleNodeWith({
+    required TreeViewNode<dynamic> node,
+    required Widget child,
+  }) {
+    return Builder(builder: (BuildContext context) {
+      return GestureDetector(
+        behavior: HitTestBehavior.translucent,
+        onTap: () {
+          TreeViewController.of(context).toggleNode(node);
+        },
+        child: child,
+      );
+    });
+  }
+
+  /// Returns the fixed height, default [TreeRow] for rows in the tree,
+  /// which is 40 pixels.
+  ///
+  /// Used by [TreeView.defaultTreeRowBuilder].
+  static TreeRow defaultTreeRowBuilder(TreeViewNode<dynamic> node) {
+    return const TreeRow(
+      extent: FixedTreeRowExtent(_kDefaultRowExtent),
+    );
+  }
+
+  /// Returns the default tree row for a given [TreeViewNode].
+  ///
+  /// Used by [TreeView.defaultTreeNodeBuilder].
+  ///
+  /// This will return a [Row] containing the [toString] of
+  /// [TreeViewNode.content]. If the [TreeViewNode] is a parent of additional
+  /// nodes, a arrow icon will precede the content, and will trigger an expand
+  /// and collapse animation when tapped based on the [TreeView.animationStyle].
+  static Widget defaultTreeNodeBuilder(
+    BuildContext context,
+    TreeViewNode<dynamic> node, {
+    AnimationStyle? animationStyle,
+  }) {
+    final Duration animationDuration =
+        animationStyle?.duration ?? TreeView.defaultAnimationDuration;
+    final Curve animationCurve =
+        animationStyle?.curve ?? TreeView.defaultAnimationCurve;
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Row(children: <Widget>[
+        // Icon for parent nodes
+        TreeView.toggleNodeWith(
+          node: node,
+          child: SizedBox.square(
+            dimension: 30.0,
+            child: node.children.isNotEmpty
+                ? AnimatedRotation(
+                    turns: node.isExpanded ? 0.25 : 0.0,
+                    duration: animationDuration,
+                    curve: animationCurve,
+                    child: const Icon(IconData(0x25BA), size: 14),
```

Comment:
Curious about this. I don't want to depend on material just for an icon, but I know developers want a basic default like this. Is there a way I can do this and keep the icon shaker happy? 
I moved SliverTree in https://github.com/flutter/flutter/pull/147171 into the material library for the same single icon. 🫤

Reply:
Confirmed in https://github.com/flutter/flutter/issues/147189 this should be ok, fixed in https://github.com/flutter/flutter/pull/147202. I don't think it will block this landing or make it wait on a stable release.

Reply:
Totally fair. 😆